acts.md

```diff
@@ -201,4 +201,13 @@ Lists the behaviours that this spec calls. If the listed behaviour's `_pass` pro
 calls
     DSToken.transferFrom
 ```
-    
+
+### where
+Allows for the definition of named expressions. All instances of the name on the LHS of the `where` binding will be expanded to the RHS as part of `klab build`. Helps to reduce duplication in specs for methods that have lots of intermediate variables.
+
+```
+where
+
+    Var => X + Y
```

Comment:
I would use := or just = instead of the rewrite arrow. The equality makes it clear that this expression is really just used as a name for the lobger expression, the rewrite arrow introduces questions like "in what order are things rewritten?"

Reply:
@MrChico if this `where` is different from the act2 `where` (e.g. cause act2 `where` is recursive), it could be smart to not use the same symbol.

Could it even be a feature to provide recursive and non-recursive assignments/substitutions?

Not sure if you're thinking of completely breaking backwards compat with act2.

Reply:
This has less to do with act2 compat and more to do with aesthetics. To me its cleaner to define the rhs as an expression rather than a term which can be further rewritten

Reply:
This has less to do with act2 compat and more to do with aesthetics. To me its cleaner to define the rhs as an expression rather than a term which can be further rewritten

Reply:
Obviously only a syntactic concern, I just find it cleaner

Reply:
Act behaviours should define the atomic steps in a transition system. Each behaviour defines one (large step) rewrite. The `where` section can be used to help expressing the rhs of this rewrite, but introducing helpers as rewrite rules themselves is too nuch rewriting maximalism for my tastes

Reply:
Not 100% sure what you mean with: `an expression rather than a term which can be further rewritten`, but the current implementation allows the RHS of macros in the `where` block to be rewritten by other macros in the `where` block. Do you see this as a problem?

Reply:
I don't see that as a problem, but I think thinking of what the `where` header does as defining further rewrites is not a perspective that I would like to proliferate

Reply:
ah ok, you want to disambiguate them from `K` rewrite rules. Yes, this makes sense.

Reply:
done.